.gitignore

```diff
@@ -1,2 +1,22 @@
-dist/
+# Build artifacts
+serial-discovery
+serial-discovery.exe
+__pycache__/
 
+# Test artifacts
+coverage_unit.txt
+
+# IDEs
+.idea/
+.vscode/
+*.bak
+*.code-workspace
+*.sublime-workspace
+
+# MkDocs
+/site/
+/docsgen/arduino-cli
+/docsgen/arduino-cli.exe
```

Comment:
Are these references to `arduino-cli` appropriate for this project?

Reply:
Not at all, I'll remove it. By the way I copied that from the Arduino Lint `.gitignore` so we should probably remove it from there too.

Reply:
Hah! As I mentioned before, I was intending to look into putting a "template" `.gitignore` into the assets repo. I think it would be nice to have that, since I often end up generating my `.gitignore` files little by little based on one annoying pollution of the repository to another, which is both tedious, and also likely to be insufficient for other contributors. It would be great to be able to just copy/paste one in to the project at the beginning that would cover all the bases.

Reply:
I use gitignore.io usually for this purposes, not sure having a template would be much more helpful.